File: android/src/main/java/top/huic/tencent_im_plugin/message/SoundMessageNode.java
package top.huic.tencent_im_plugin.message;

import com.tencent.imsdk.v2.V2TIMManager;
import com.tencent.imsdk.v2.V2TIMMessage;
import com.tencent.imsdk.v2.V2TIMSoundElem;

import top.huic.tencent_im_plugin.message.entity.SoundMessageEntity;

/**
 * 语音消息节点
 */
public class SoundMessageNode extends AbstractMessageNode<V2TIMSoundElem, SoundMessageEntity> {

    @Override
    public V2TIMMessage getV2TIMMessage(SoundMessageEntity entity) {
        return V2TIMManager.getMessageManager().createSoundMessage(entity.getPath(), entity.getDuration());
    }

    @Override
    public String getNote(V2TIMSoundElem elem) {
        return "[语音]";
    }

    @Override
    public SoundMessageEntity analysis(V2TIMSoundElem elem) {
        return new SoundMessageEntity(elem);
    }

    @Override
    public Class<SoundMessageEntity> getEntityClass() {
        return SoundMessageEntity.class;
    }
}

File: ios/Classes/message/SoundMessageNode.swift
import ImSDK

//
//  TextMessageNode.swift
//  tencent_im_plugin
//
//  Created by 蒋具宏 on 2020/3/13.
//  语音消息节点
public class SoundMessageNode: AbstractMessageNode {

    override func getV2TIMMessage(params: [String: Any]) -> V2TIMMessage {
        let path: String = getParam(params: params, paramKey: "path")!;
        let duration: Int32 = getParam(params: params, paramKey: "duration")!;
        return V2TIMManager.sharedInstance().createSoundMessage(path, duration: duration)
    }

    override func getNote(elem: V2TIMElem) -> String {
        "[语音]";
    }

    override func analysis(elem: V2TIMElem) -> AbstractMessageEntity {
        SoundMessageEntity(elem: elem as! V2TIMSoundElem);
    }
}
